### PR TITLE
Create reusable components

### PR DESCRIPTION
**Pull request by Briantam0422**
I've created an Event List component and an Event Item component that can be used on "My Event", event browsing page, event overview page, and user dashboard.

you can find the demo in route **"/ui-components/event-components-demo"**

Style can be adjusted after the design comes out.

Done:
1. Event List component -> for displaying multiple events
2. Event Item component -> The design of an event
3. Event Item Status component -> for showing different status with different color
4. Demo page for showing Events
5. uploaded a demo image for the showcase
6. A list of components on "ui-components" page. Helps developers to find components easily.

Features: 
1. Responsive
2. Reusable

image

image

image

### CODE REVIEW

File: src/components/Event/EventList.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import EventListItem, { iEventItem } from "./EventItem";
+
+interface EventListProps {
+  events: Array<iEventItem>;
+}
+
+export const EventList = (props: EventListProps) => {
+  return (
+    <>
```
Comment (ialej001) [inline src/components/Event/EventList.tsx:9]: <></> not required here. <div></div> is the parent element and it's required for layout/style.

File: src/pages/ui-components/button-demo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+import { FaDiscord } from "react-icons/fa";
+import Button from "~/components/Button";
+
+const ButtonDemo = (props) => {
+  const onClick = () => {
+    console.log("I do nothing");
+  };
+  return (
+    <>
```
Comment (ialej001) [inline src/pages/ui-components/button-demo/index.tsx:11]: <></> not required.

File: src/components/Event/EventList.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      <div className="md:10 lg:20 flex grid grid-cols-1 flex-wrap gap-5 p-5 md:grid-cols-2 lg:grid-cols-3">
+        {props.events.map((event) => {
+          return (
+            <>
```
Comment (ialej001) [inline src/components/Event/EventList.tsx:13]: <></> not required. It's already a single component being returned.

File: src/components/Event/EventItem.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import { EventList } from "./EventList";
+import Image from "next/image";
+import { CiCalendarDate, CiBookmarkCheck } from "react-icons/ci";
+import { HiOutlineTicket } from "react-icons/hi2";
+import { MdOutlinePublic } from "react-icons/md";
+import { RiGhostSmileLine } from "react-icons/ri";
+import { MouseEventHandler } from "react";
+import { EventItemStatus } from "./EventItemStatus";
+
+export interface iEventItem {
```
Comment (ialej001) [inline src/components/Event/EventItem.tsx:11]: I believe in TypeScript we don't identify interfaces with this naming convention, especially when the interface is being used to describe props for a component. I believe it should be named `EventItemProps`, but you don't have to change it right now. We might/likely would refactor this and we can rename it then

**Review by ialej001 — APPROVED** [review]
I left some comments here and there. I won't require changes, but keep it in mind going forward on the use of <></> versus <div></div> and when it is appropriate to use either when returning JSX.

**Review by mpark4656 — APPROVED** [review]
Looks great!

File: src/components/Event/EventItemStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      break;
+    case "COMPLETED":
+      bgColor += "bg-green-600";
+      break;
+    case "IN_PROGRESS":
+      bgColor += "bg-blue-600";
+      break;
+  }
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (Briantam0422) [reply src/components/Event/EventItemStatus.tsx:25]: Got it! I will keep it in mind. Thanks

File: src/components/Event/EventItem.tsx (hunk 4 above)
Comment (Briantam0422) [reply src/components/Event/EventItem.tsx:11]: Got it!

**Pull request merged**